### ISSUE TITLE
is_chunked function?

### ISSUE THREAD

**Issue by Luapulu** [JuliaIO/HDF5.jl#727]
Is there a function to check that a dataset is or is not chunked? I've been using a try statement around the `get_chunk` function, but I'd like to be able to check directly without having the whole c error message get printed every time I check for chunking.

**Comment by jmert** [comment]
It looks like `https://portal.hdfgroup.org/display/HDF5/H5P_GET_LAYOUT` is the function to use. Wrapping that can be easily added.

**Comment by Luapulu** [comment]
Awesome, thanks. I'll make a PR when I get to it.

**Comment by musm** [comment]
```julia
function ischunked(obj::Dataset)
    prop = h5d_get_create_plist(checkvalid(obj))
    try
        h5p_get_layout(prop) == H5D_CHUNKED
    finally
        h5p_close(prop)
    end
end
```
Should we just add this as a function, or is there a more clever way to handle all cases without three different functions for testing  which one of  `H5D_CONTIGUOUS, H5D_CHUNKED,H5D_COMPACT` is true.

https://github.com/HDFGroup/hdf5/blob/58d8eae182ac764ef15c3ed040a12a96aa4d16f0/src/H5Dpublic.h#L43

**Comment by Luapulu** [comment]
How about also adding a convenience `h5p_get_layout` method similar to `get_chunk`, that directly applies to some `obj`?

**Comment by jmert** [comment]
> Should we just add this as a function, or is there a more clever way to handle all cases without three different functions for testing which one of `H5D_CONTIGUOUS, H5D_CHUNKED,H5D_COMPACT` is true.

I'm not opposed to adding several functions (given that `iscontiguous` already exists), but in the spirit of #699's idea to unexport more functions, these should probably all be unexported public functions.

The other option is to add just a `get_layout` function and just state that the result should be compared to the appropriate constants. (This would probably be a more attractive option if/when some of the constants are refactored to `@enum`s.)

**Comment by musm** [comment]
That sounds like a reasonable plan.

In other words, add `ischunked` and `iscompact`   (I think `H5D_VIRTUAL` may not be available on v1.10, haven't checked)

As a follow up add `get_layout` with the appropriate modifications to the constants (use enums to make the interface simpler to use)

**Comment by musm** [comment]
closed by https://github.com/JuliaIO/HDF5.jl/pull/740
